### PR TITLE
feat: 🎸 Added print area edit

### PR DESCRIPTION
**Pull request by krishnajithr5**
✅ Closes: #24

### CODE REVIEW

File: custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js [hunk 1]
```diff
@@ -1,13 +1,99 @@
...
+      case "create":
+        this.savePrintAreas();
+        break;
+      case "edit":
+        this.editPrintAreas();
+        break;
+      default:
+        this.getPrintAreas();
+    }
+  }
```
Comment (harikrishna266) [inline custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js:22]: nice

File: custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.php [hunk 2]
```diff
@@ -10,4 +10,41 @@ public function get_print_area_data()
...
+    }
+
+    public function edit_print_area_data()
+    {
+        $height = isset($_POST['height']) ? $_POST['height'] : '';
+        $width = isset($_POST['width']) ? $_POST['width'] : '';
+        global $wpdb;
+        $table_name = $wpdb->prefix . 'print_areas';
+        $wpdb->update(
+            $table_name,
```
Comment (harikrishna266) [inline custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.php:37]: add some kind of checks to make sure that no special characters are getting inserted

File: custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js [hunk 3]
```diff
@@ -1,13 +1,99 @@
...
+        if (xhr.status >= 200 && xhr.status < 300) {
+          this.redirectToPrintingAreasBuilder();
+        } else {
+        }
+      };
+      xhr.send(formData);
+    });
+  }
+
+  async editPrintAreas() {
```
Comment (harikrishna266) [inline custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js:63]: we do not need api for edit.

File: custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js (hunk 1 above)
Comment (harikrishna266) [reply custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js:22]: now that we don't use event listeners we can move this code to BE.

File: custom-plugins/fictive_studios/includes/class-builder.php [hunk 4]
```diff
@@ -52,6 +52,8 @@ private function printing_areas_CRUD(){
 
         $printing_area_admin_api = new PrintAreaAPIAdmin;
         $this->loader->add_action( 'wp_ajax_get_print_areas', $printing_area_admin_api, 'get_print_area_data' );
+        $this->loader->add_action( 'wp_ajax_save_print_areas', $printing_area_admin_api, 'save_print_area_data' );
+        $this->loader->add_action( 'wp_ajax_edit_print_areas', $printing_area_admin_api, 'edit_print_area_data' );
```
Comment (harikrishna266) [inline custom-plugins/fictive_studios/includes/class-builder.php:56]: remove these hooks as we are not using ajax calls

**Comment by harikrishna266** [comment]
add a few dummy values into this table

File: custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.php [hunk 5]
```diff
@@ -10,4 +10,41 @@ public function get_print_area_data()
         wp_die();
     }
 
+    public function save_print_area_data()
+    {
+        $height = isset($_POST['height']) ? $_POST['height'] : '';
+        $width = isset($_POST['width']) ? $_POST['width'] : '';
+        global $wpdb;
+        $table_name = $wpdb->prefix . 'print_areas';
+        $wpdb->insert(
```
[the remark this answers is not in the document]
Comment (krishnajithr5) [reply custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.php:19]: `$wpdb->insert` and `$wpdb->update` will check for sql injection automatically, `$wpdb->prepare` is not needed.
[https://stackoverflow.com/questions/18096555/how-to-insert-data-using-wpdb](url)

File: custom-plugins/fictive_studios/admin/printing-area/create/print-area-create-view.php [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        <button id="savePrintAreaBtn" type="submit">Submit</button>
+      </div>
+    </div>
+  </div>
+</form>
+
+<?php
+echo <<<HTML
+<style>
+  #print-area-form {
```
Comment (harikrishna266) [inline custom-plugins/fictive_studios/admin/printing-area/create/print-area-create-view.php:35]: can't we use the same classes that are used in wordpress

File: custom-plugins/fictive_studios/admin/printing-area/create/print-area-create.php [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<?php
+namespace FictiveCodes;
+
+class PrintAreaCreate
+{
+    public function get_create_print_area_template()
+    {
+        $action = ($_GET['action']);    
+        $print_area_id;
```
Comment (harikrishna266) [inline custom-plugins/fictive_studios/admin/printing-area/create/print-area-create.php:9]: do we need this?

File: custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js [hunk 8]
```diff
@@ -1,13 +1,17 @@
...
 
-function getPrintAreas() {
-  var xhr = new XMLHttpRequest();
-  var params = new URLSearchParams();
-  params.append("action", "get_print_areas");
-  var url = ajaxurl + "?" + params.toString();
-  xhr.open("GET", url, true);
-  xhr.onload = function () {};
-  xhr.send();
+  getPrintAreas() {
```
Comment (harikrishna266) [inline custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js:4]: Remove this.

Comment (harikrishna266) [reply custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js:4]: I think you can remove the whole file.

Comment (krishnajithr5) [reply custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js:4]: Don't you need the api for builder?

File: custom-plugins/fictive_studios/admin/printing-area/create/print-area-create.php (hunk 7 above)
Comment (krishnajithr5) [reply custom-plugins/fictive_studios/admin/printing-area/create/print-area-create.php:9]: I use those variables to retrieve print area data info based upon the url parameters if the page in in edit state.

File: custom-plugins/fictive_studios/admin/printing-area/create/print-area-create-view.php [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+      </tr>
+      <tr class="form-field">
+        <th scope="row"></th>
+        <td>
+          <button id="savePrintAreaBtn" type="submit" class="button button-primary">Submit</button>
+        </td>
+      </tr>
+    </tbody>
+  </table>
+</form>
```
Comment (krishnajithr5) [inline custom-plugins/fictive_studios/admin/printing-area/create/print-area-create-view.php:33]: Removed the custom styling. Now uses default wordpress admin style format.
https://dotorgstyleguide.wordpress.com/

File: custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js (hunk 8 above)
Comment (harikrishna266) [reply custom-plugins/fictive_studios/admin/printing-area/api/print-area-api.js:4]: yes.. but this code is not needed here..

**Review by harikrishna266 — APPROVED** [review]

**Pull request merged**